python/bindings.cpp

```diff
@@ -299,7 +299,10 @@ PYBIND11_MODULE(libsonata, m)
             "Selection from list of intervals"
         )
         .def(
-            py::init(&Selection::fromValues),
+            py::init([](py::array_t<uint64_t, py::array::c_style | py::array::forcecast> values) {
```

Comment:
I have not found a "canonical" way to get (begin, end) iterator pair from NumPy array object in `pybind11`.
I _hope_ that casting to C-style array and using raw pointers as iterator pair is safe.
Alternatively, we can make an iterator class ourselves but that seems to be a bit of an overkill.

Reply:
isn't there a chance that the NumPy array isn't contiguous: ie if it's a slice into another array, the iteration may need to have a strides setup.

I think it's safe to do what you're doing, as long as there's a way to check for contiguousness, and fall back to the slow path otherwise.

Reply:
> isn't there a chance that the NumPy array isn't contiguous

As far as I understand, that's what `py::array::forcecast` is aimed to ensure:
https://pybind11.readthedocs.io/en/stable/advanced/pycpp/numpy.html#arrays

I haven't tried with a strided array, guess it's worth checking.

Reply:
Ah, thanks.  That sounds like it does the right thing, sorry about the noise.

Reply:
On contrary, it's easy to get trapped by false assumptions with this stuff.
I have added some unit tests at #30.
Truth be told, I'm still not 100% sure it does the right thing. ^^